src/Temporalio/Worker/WorkflowInstance.cs

```diff
@@ -191,7 +243,9 @@ public WorkflowActivationCompletion Activate(WorkflowActivation act)
             {
                 completion = new() { RunId = act.RunId, Successful = new() };
                 currentActivationException = null;
-                // We need to sort jobs
+                IsReplaying = act.IsReplaying;
+                UtcNow = act.Timestamp.ToDateTime();
+                // We need to sort jobs by notify, then signal, then non-query, then query
                 var jobs = act.Jobs.OrderBy(job =>
```

Comment:
nit: don't need this if you upgrade core, can get rid of it in the next PR

Reply:
Yeah, there's a few things I have to change on next core upgrade. I am delaying it a bit while in mid-development on other things.